src/services/bundle_repository.js

```diff
@@ -78,14 +80,33 @@ export default class BundleRepository {
     return await downloadJSONFromGridFSBucket(bundleId, this.bundlesBucket);
   }
 
-  async createBundleMetadata(bundleId, storagePeriods, status = BundleStatusStates.unknown) {
+  async createBundleMetadata(bundleId, storagePeriods, status = BundleStatuses.unknown, additionalRepositoryFields = {}) {
     if (await this.db.collection('bundle_metadata').findOne({bundleId}) === null) {
       await this.db.collection('bundle_metadata').insertOne({
-        bundleId, storagePeriods, repository: {status}
+        bundleId, storagePeriods, repository: {...additionalRepositoryFields, status}
       });
     }
   }
 
+  async storeAdditionalMetadata(bundleId, complementaryMetadata) {
+    const requiredFields = [
```

Comment:
I imagine those fields could simply come from the initialMetadata. So in the data model engine we could have sth like: `bundle_repository.updateBundleMetadata(bundleId, additionalBundleMetadataFields(initialMetadata, downloadedMetadata))`